rspace/src/main/scala/coop/rchain/rspace/history/LMDBTrieStore.scala

```diff
@@ -48,6 +51,22 @@ class LMDBTrieStore[K, V] private (val env: Env[ByteBuffer], _dbTrie: Dbi[ByteBu
       Codec[Trie[K, V]].decode(BitVector(buffer)).map(_.value).get
     }
   }
+
+  private[rspace] def toMap: Map[Blake2b256Hash, Trie[K, V]] =
+    withTxn(createTxnRead()) { txn =>
+      val keyRange: KeyRange[ByteBuffer] = KeyRange.all()
+      withResource(_dbTrie.iterate(txn, keyRange)) { (it: CursorIterator[ByteBuffer]) =>
```

Comment:
You are iterating over all keys here. There's a method for that which doesn't need the key range:
```
/**
   * Iterate the database from the first item and forwards.
   *
   * @param txn transaction handle (not null; not committed)
   * @return iterator
   */
  public CursorIterator<T> iterate(final Txn<T> txn) {
    return iterate(txn, all());
  }
```

Reply:
Oh fantastic.  Thanks for pointing that out!